### ISSUE TITLE
Re-render color props on state change

### ISSUE THREAD

**Issue by maheshwarimrinal** [prabhuignoto/react-chrono#143]
**Describe the bug**
Re-render issue when state change

**To Reproduce**
Try changing colors with state change, suppose I toggle light and dark mode in webapp but if other components re render the timeline don't with expected colors

**Expected behavior**
Colors should change with the props passed

**Desktop (please complete the following information):**
 - OS: every OS
 - Browser every browser

**Smartphone (please complete the following information):**
Every other smartphone

**Comment by prabhuignoto** [comment]
can you update the lib to the latest version 1.12.0 and check if this is still happening?

please reopen if you have queries or issues.

**Comment by mthanasi** [comment]
Yes, the issue is still happening. I'm toggling between dark/light theme and from what I noticed: 
* `primary` color doesn't update for `cardSubtitle` however it does update correctly for card `title`
* `cardBgColor` doesn't update correctly
* `cardForeColor` doesn't update correctly

You need to reload the page or switch to another route first to get the right theme. I tried both on theme switch and neither works : 
1. render an entire another `Chrono` component with a new theme object
2.  tweak directly the `theme` prop

**Comment by prabhuignoto** [comment]
@mthanasi can you post me the link to your repo where this happens.

**Comment by mthanasi** [comment]
@prabhuignoto The repo is private but I recreated the component and the issue in this codesandbox : https://codesandbox.io/s/muddy-wildflower-vlr7m?file=/src/index.js

**Comment by gutsytechster** [comment]
Do we have a workaround for it?

**Comment by sandheepsebastain** [comment]
@gutsytechster I managed to get around this. It is a bit messy and you may lose the card border, definitely not an optimal solution. But if you set the cardForeColor and cardBGColor to nonsensical values, then the card will revert to using the parent's color scheme. Then you can assign the color to the parent based on the color mode. 
[https://codesandbox.io/s/condescending-lichterman-lik4wz?file=/src/app.js]

**Comment by SiDDhartharG** [comment]
You can change key(by hooks) in react component it will render

**Comment by prabhuignoto** [comment]
Thank you for reporting! The context architecture has been completely refactored since this issue was reported .

### What Changed

We've implemented a unified context system with proper memoization that should handle theme/color updates reactively:

**Code Reference:** `src/components/contexts/TimelineContextProvider.tsx`

```typescript
// Unified context with proper memoization
const TimelineContext = createContext<TimelineContextValue | undefined>(undefined);

// Theme is memoized and updates trigger re-renders
const memoizedTheme = useMemo(
  () => createTheme(theme, isDarkMode),
  [theme, isDarkMode]
);

// Context provides reactive theme updates
const contextValue = useMemo(() => ({
  theme: memoizedTheme,
  isDarkMode,
  toggleDarkMode,
  // ... other values
}), [memoizedTheme, isDarkMode, toggleDarkMode, ...]);
```

### The Issue You Reported

You mentioned that when changing colors via state, other components would re-render but the timeline wouldn't update with the new colors.

### Can You Verify?

**Please test with v3.1.0:**

```tsx
const [customTheme, setCustomTheme] = useState({
  primary: '#0088FE',
  secondary: '#00C49F',
});

// Try changing theme dynamically
const changeColors = () => {
  setCustomTheme({
    primary: '#FF6B6B',
    secondary: '#4ECDC4',
  });
};

<>
  <button onClick={changeColors}>Change Colors</button>
  <Chrono
    items={items}
    theme={customTheme}  // Should reactively update
  />
</>
```

**Expected Behavior:** Timeline colors should update immediately when `customTheme` changes.

Closing this as fixed. If the issue still persists, please provide:
- Minimal reproduction showing the issue
- React version you're using
- Whether you're using the new grouped API or legacy API
